packages/plugin/src/theia.d.ts

```diff
@@ -9574,14 +9591,20 @@ declare module '@theia/plugin' {
         export function registerDebugAdapterDescriptorFactory(debugType: string, factory: DebugAdapterDescriptorFactory): Disposable;
 
         /**
-         * Register a [debug configuration provider](#DebugConfigurationProvider) for a specific debug type.
+         * Register a {@link DebugConfigurationProvider debug configuration provider} for a specific debug type.
+         * The optional {@link DebugConfigurationProviderTriggerKind triggerKind} can be used to specify when the `provideDebugConfigurations` method of the provider is triggered.
+         * Currently two trigger kinds are possible: with the value `Initial` (or if no trigger kind argument is given) the `provideDebugConfigurations` method is used to provide the initial debug configurations to be copied into a newly created launch.json.
+         * With the trigger kind `Dynamic` the `provideDebugConfigurations` method is used to dynamically determine debug configurations to be presented to the user (in addition to the static configurations from the launch.json).
```

Comment:
I think the explanations elsewhere are clearer: 

```suggestion
         * With the trigger kind `Dynamic` the `provideDebugConfigurations` method is called to dynamically determine debug configurations to be presented when the user requests all available configurations from the UI.
```

Reply:
Done! I have Aligned it with the rest of the explanations.